lib/coinbase/wallet.rb

```diff
@@ -223,22 +246,38 @@ def self.from_hash(data)
 
     private
 
-    # Derives an already registered Address in the Wallet.
-    # @return [Address] The new Address
-    def derive_address
-      key = derive_key
+    # Derives the registered Addresses in the Wallet.
+    # @param address_models [Array<Coinbase::Client::Address>] The models of the addresses already registered with the
+    #   Wallet
+    def derive_addresses(address_models)
+      # Create a map tracking which addresses are already registered with the Wallet.
+      address_map = build_address_map(address_models)
```

Comment:
Is this address map actually used anywhere? It looks like we do not used it in `def derive_address`?

Reply:
This seems like we're just including a map  to make sure that we're deriving keys in the appropriate order?

Is this because we don't expect (or want to rely on) addresses being result in address index order? 

Otherwise it seems like we could just iterate through the address models and derive key for that specific index